### ISSUE TITLE
Pick up todo items in phpdoc

### ISSUE THREAD

**Issue by christeredvartsen** [Gruntfuggly/todo-tree#228]
When adding multiline docs to PHP functions, the "standard" is to do this:

```php
/**
 * @todo Remember to do something
 */
function somefunc() {
}
```

The `@todo` tag is documented [here](https://docs.phpdoc.org/references/phpdoc/tags/todo.html).

This is not picked up by the default configuration of Todo Tree, but a small change could fix this. The `todo-tree.general.tags` would have to include `@todo` and `todo-tree.regex.regex` would have to make the `/` in `/*` optional, like this:

```json
{
    "todo-tree.regex.regex": "((//|#|<!--|;|/?\\*|^)\\s*($TAGS)|^\\s*- \\[ \\])",
    "todo-tree.general.tags": [
        "@todo",
        "TODO",
        "FIXME"
      ]
}
```

**Comment by Gruntfuggly** [comment]
Hi - thanks for the pull request, but I stopped making changes to the default regex a while ago. It tended to cause problems for people who were using the previous default.

I am going to add a wiki page soon and capture the suggestions people have made, so I will include this one.

**Comment by christeredvartsen** [comment]
Perfect, thanks.

Would it be possible to have Todo-Tree support custom regular expressions that will be executed in addition to the default one? So that I won't have to change the default regular expression at all, but simply run my own in addition.

**Comment by Gruntfuggly** [comment]
You could add yours as an alternate - the default currently has a regex which matches tags, and an alternate which matches markdown TODO's, separated with a vertical bar. You could add yours as a third, e.g.
`((//|#|<!--|;|/\*|^)\s*($TAGS)|^\s*- \[ \]|<yours here>)`

**Comment by christeredvartsen** [comment]
Yes, that is what I currently do, but what I would like is to not touch the default at all, and simply add my own regular expression to be run in addition.

I know this is currently not supported, but having a list of additional expressions to be run as well might be a nice feature addition to Todo-Tree. This makes it easier to add custom stuff, without having to touch the default.

**Comment by Gruntfuggly** [comment]
I'm curious - what is the benefit of not changing the default? It could be done, but it would add more complexity, and I don't understand what you would gain from it?

**Comment by christeredvartsen** [comment]
> what is the benefit of not changing the default?
I usually think it's safer to leave default settings alone, so that the maintainer of the software can tweak and update the settings. But if you say that the defualt expression won't be touch that much I guess it's safe enough for me to change it locally.

**Comment by Gruntfuggly** [comment]
Ah OK - in this case, the default is really just something to get people started - I imagined everybody would need to tweak it for they're own use. I don't plan to change it anymore unless somebody finds a problem with it.

**Comment by cshanejennings** [comment]
Slight change from what I found in the other issues, but for anyone stumbling on this in a search, there's two changes needed in settings

**Settings** > **Extensions** > **ToDo Tree** > **General: Tags**
add `@todo` to make @todo a recognized tag

**Settings** > **Extensions** > **ToDo Tree** > **General: Tags** > **Regex: Regex**
set the regex to  `(//|#|<!--|;|/?\*|^|^[ \t]*(-|\d+.))\s*($TAGS)` the added "?\*" sets the regex to ignore the leading * in PHPDoc comments, so it can pick up the @todo line
